beacon-chain/sync/backfill/verify.go

```diff
@@ -0,0 +1,113 @@
+package backfill
+
+import (
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/core/signing"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/state"
+	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v4/config/params"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/blocks"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/interfaces"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v4/crypto/bls"
+	"github.com/prysmaticlabs/prysm/v4/network/forks"
+	"github.com/prysmaticlabs/prysm/v4/time/slots"
+)
+
+var errInvalidBatchChain = errors.New("parent_root of block does not match root of previous")
+var errProposerIndexTooHigh = errors.New("proposer index not present in origin state")
+var errUnknownDomain = errors.New("runtime error looking up signing domain for fork")
+
+// VerifiedROBlocks represents a slice of blocks that have passed signature verification.
+type VerifiedROBlocks []blocks.ROBlock
+
+type verifier struct {
+	// chkptVals is the set of validators from the state used to initialize the node via checkpoint sync.
+	keys        [][fieldparams.BLSPubkeyLength]byte
+	maxVal      primitives.ValidatorIndex
+	vr          []byte
+	fsched      forks.OrderedSchedule
+	dt          [bls.DomainByteLength]byte
```

Comment:
It was hard for me to figure out what `dt` means. Can you rename it to `domain`?

Reply:
Short for [DomainType](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#custom-types). `domain` would be a confusing name, because the domain is derived jointly from the DomainType, Fork Version, and Genesis Validator Root. Maybe `dtype`?

Reply:
In general I think that the meaning of a value can be made clear by context. This value is initialized like `dt:          params.BeaconConfig().DomainBeaconProposer` (where the value assigned is a DomainType) and passed to ComputeDomain (where it's known to be the DomainType).